src/v2/guide/migration-vue-router.md

```diff
@@ -236,6 +236,23 @@ if (route.meta.requiresAuth) {
 </div>
 {% endraw %}
 
+### [] Syntax for Array of Route Query <sup>removed</sup>
+
+In URL, [] syntax for array of query is removed. For example, `/foo?users[]=Tom&users[]=Jerry` will be `/foo?users=Tom&users=Jerry` now.
+
+Since vue-router will treat `?users=Tom` as `query.users == 'Tom'`, unless you are for sure that the array will have more than one element, you should check and convert the query to array if necessary.
+
+One practical way is to add the following kind of watcher:
+
+```javascript
+'$route.query.users': {
+  handler(val) {
+    this.$route.query.users = Array.isArray(val) ? val : [val]
```

Comment:
We shouldn't talk about watchers, simply show the example like:

```js
const users = Array.isArray(this.$route.query.users)
  ? users
  : [users]
```

Modifying the $route object is a bad idea because it's like a snapshot of the current state, it's supposed to be read only